src/components/form/__internal__/form-summary.component.tsx

```diff
@@ -1,15 +1,31 @@
 import React, { useMemo } from "react";
-import PropTypes from "prop-types";
 
 import {
   StyledFormSummary,
   StyledInternalSummary,
   StyledMessagePrefix,
+  StyledInternalSummaryProps,
 } from "./form-summary.style";
 import Icon from "../../icon";
 import useLocale from "../../../hooks/__internal__/useLocale";
 
-const Summary = ({ type, errors, warnings }) => {
+interface SummaryProps extends StyledInternalSummaryProps {
+  errors?: number;
+  warnings?: number;
+}
+
+export interface FormSummaryProps {
+  /** Child elements */
+  children?: React.ReactNode;
+  /** The total number of errors present in the form */
+  errors?: number;
+  /** The total number of warnings present in the form */
+  warnings?: number;
+  /** Applies full width styling */
+  fullWidth?: boolean;
+}
+
+export const Summary = ({ type, errors = 0, warnings = 0 }: SummaryProps) => {
```

Comment:
What do you think about refactoring the `type` parameter to have type: `"error" | "warning"` instead of the plural forms? That way we can avoid using the type assertion below.